tools/dups/src/main.rs

```diff
@@ -348,27 +347,20 @@ fn do_dups_report(output_file: Option<String>, threshold: f64) {
 
     for pair in pairs.clone() {
         let dir = pair.asm_dir;
-        let mut funcs = Vec::new();
-        process_directory(&dir, &mut funcs);
-
-        // add matchings dir as well to get decompiled funcs
-        let matchings = dir.replace("nonmatchings", "matchings");
-        process_directory(&matchings, &mut funcs);
-
-        // sort functions by vram address
-        funcs.sort_by_key(|function| {
-            function
-                .ops
-                .first()
-                .map_or(u64::MAX, |instr| instr.vram_addr)
-        });
+        process_asm_directory(&dir, &mut files);
 
-        files.push(DupsFile {
-            name: dir.to_string(),
-            funcs: funcs.clone(),
-        });
+        let nonmatchings = dir.replace("matchings", "nonmatchings");
+        process_asm_directory(&nonmatchings, &mut files);
     }
 
+    files.sort_by(|a, b| {
+        // First, sort by whether the name contains "nonmatching" (false before true)
+        a.name
+            .contains("nonmatching")
+            .cmp(&b.name.contains("nonmatching"))
+            .then_with(|| a.name.cmp(&b.name)) // Then, sort alphabetically by name
+    });
```

Comment:
I think this should also give more stable ordering to the duplicate report since it's now alphabetical